packages/autoloader/src/class-version-selector.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/* HEADER */ // phpcs:ignore
+
+/**
+ * Used to select package versions.
+ */
+class Version_Selector {
+
+	/**
+	 * Indicates whether a package class or file's plugin has a development version.
+	 *
+	 * @var Array
+	 */
+	private static $dev_plugin_in_use = array();
+
+	/**
+	 * Checks whether the selected package version should be updated.
+	 *
+	 * @param String $class The class name.
+	 * @param String $selected_version The currently selected package version.
+	 * @param String $compare_version The package version that is being evaluated to
+	 *                                determine if the version needs to be updated.
+	 * @param String $plugin The name of the plugin that contains the package.
+	 *
+	 * @return Boolean Returns true if the selected package version should be updated,
+	 *                 else false.
+	 */
+	public function update_package_version( $class, $selected_version, $compare_version, $plugin ) {
```

Comment:
What do you say if we rename this method to something that indicates that it returns a boolean, but doesn't actually update the version. Like `is_version_update_required` , or something of the sort. With the current name I was expecting this method to actually do something instead.

Reply:
I agree, and I'll change the method name.

Reply:
Done in commit cd9bf72